sql/core/src/test/scala/org/apache/spark/sql/sources/BucketedReadSuite.scala

```diff
@@ -735,4 +744,88 @@ abstract class BucketedReadSuite extends QueryTest with SQLTestUtils {
         df1.groupBy("j").agg(max("k")))
     }
   }
+
+  //  a test with a single bucketed partition where the number of files in the partition is large
+  //  tests for the condition where the serialization of such a task may result in a stack overflow
+  //  if the files list is stored in a recursive data structure
+  test("SPARK-27100 stack overflow: read bucketed data with large partitions") {
+    testLargeFilePartitionStackOverflow(true)
+  }
+
+  //  a test with a single non-bucketed partition where the number of files in the partition is
+  //  large tests for the condition where the serialization of such a task may result in a stack
+  //  overflow if the files list is stored in a recursive data structure
+  test("SPARK-27100 stack overflow: read non bucketed data with large partitions") {
+    testLargeFilePartitionStackOverflow(true)
```

Comment:
Should one of these be false? how about just one test that calls this for true, false? You don't need three methods then, just one.

Reply:
Yes you're right. Careless mistake on my part. And as it turns out the unit test was broken for the non-bucketed case. Fixed and unified the test cases into one.